spec/requests/graphql/mutations/term/update_term_spec.rb

```diff
@@ -0,0 +1,195 @@
+      end
+    end
+
+    context 'when adding multiple alt_labels' do
+      let(:term) { FactoryBot.create(:external_term, vocabulary: vocabulary) }
+      let(:variables) do
+        {
+          input: {
+            vocabularyStringKey: vocabulary.string_key,
+            uri: term.uri,
+            altLabels: ['Uni', 'Horse with Horn']
+          }
+        }
+      end
+      let(:expected_response) do
+        %(
+          {
+            "term": {
+              "uri": "http://id.worldcat.org/fast/1161301/",
+              "prefLabel": "Unicorns",
+              "altLabels": ["Uni", "Horse with Horn"],
+              "authority": "fast",
+              "termType": "external",
+              "customFields": [
+                { "field": "classification", "value": null },
+                { "field": "harry_potter_reference", "value": true }
+              ]
+            }
+          }
+        )
+      end
+
+      before { graphql query, variables }
+
+      it 'adds alt_labels for term' do
+        term.reload
+        expect(term.alt_labels).to match_array ['Uni', 'Horse with Horn']
+      end
+
+      it 'returns updated term' do
+        expect(response.body).to be_json_eql(expected_response).at_path('data/updateTerm')
+      end
+    end
+
+    context 'when updating term_type' do
+      let(:term) { FactoryBot.create(:external_term, vocabulary: vocabulary) }
+      let(:variables) do
+        {
+          input: {
+            vocabularyStringKey: vocabulary.string_key,
+            uri: term.uri,
+            termType: 'local'
+          }
+        }
+      end
+
+      before { graphql query, variables }
+
+      it 'does not update record' do
+        term.reload
+        expect(term.uri).not_to eql 'local'
+      end
+    end
+
+    context 'when attempting to update a uri that is not associated with a known term' do
+      let(:variables) do
+        {
+          input: {
+            vocabularyStringKey: vocabulary.string_key,
+            uri: 'not/known',
+            prefLabel: 'New Label'
+          }
+        }
+      end
+
+      before { graphql query, variables }
+
+      it 'return error' do
+        expect(response.body).to be_json_eql(%("Couldn't find Term")).at_path('errors/0/message')
+      end
+    end
+
+    context 'when updating a term on a locked vocabulary' do
+      let(:term) { FactoryBot.create(:external_term, vocabulary: vocabulary) }
+      let(:variables) do
+        {
+          input: {
+            vocabularyStringKey: vocabulary.string_key,
+            uri: term.uri,
+            altLabels: ['Uni']
+          }
+        }
+      end
+
+      before do
+        vocabulary.update(locked: true)
+        graphql query, variables
+      end
+
+      it 'returns error in json' do
+        expect(response.body).to be_json_eql(%(
+          "Vocabulary is locked."
+        )).at_path('errors/0/message')
+      end
+    end
+  end
+
+  def query
```

Comment:
Starting to wonder if we should move all these gql literals into a spec helper/support file in a future ticket.

Reply:
I actually like the queries rights in the testing spec for them. We usually never use these queries in two different specs.

Reply:
Definitely useful as long as that's true! I thought there were some repeated in this PR, but I might have misread.